### ISSUE TITLE
bug: Setting output ports at 4DI4DO with CPX-AP fails

### ISSUE THREAD

**Issue by jhildenbrand-festo** [Festo-se/festo-cpx-io#23]
***In GitLab by @chrichrichri on Aug 19, 2024, 20:33 UTC:***

Maybe it is also a configuration issue (no idea what could be wrong though), but with the most recent version (0.5.4) it is not possible for me to set the output ports at my 4DI4DO.

How to reproduce: Run [sample code](https://gitlab.com/festo-research/electric-automation/festo-cpx-io/-/blob/114fd6b820252f6225b2c6945b95f332cf93f45e/examples/example_cpxap_digital_output.py) 

What happens: Code runs. If output are read within the code the correct values are shown, anyhow the outputs are not actually set (LED do never turn on). Reading inputs works just as expected.

When toggling the outputs with Festo Automation Suite everything works as expected

*Migrated from GitLab: https://gitlab.com/festo-research/electric-automation/festo-cpx-io/-/issues/23*

**Comment by jhildenbrand-festo** [comment]
***In GitLab by @mw_festo on Aug 22, 2024, 07:03 UTC:***

Hi @chrichrichri, 

thank you for your report. This might be an issue of your eyes not beeing fast enough to see the output LEDs turn on before they turn off again when the program runs through. If the channels are read back correctly (including the output channels) they should also switch on the LEDs! 

To make the outputs stay on longer, you can add a sleep time from the time module to delay the program. Keep in mind that there is a default modbus connection timeout that will close the connection after 100 ms. You can disable that timeout completely by setting timeout=0 in the CpxAp object instantiation. I suggest trying the code below and see if the output LEDs are turning on now. They should turn on and off 10 times with a frequency of around 1 Hz. By the way, you can also use the toggle function for that usecase :smile: 

```
import time
from cpx_io.cpx_system.cpx_ap.cpx_ap import CpxAp

# use timeout=0 to disable the modbus timeout
with CpxAp(ip_address="192.168.1.1", timeout=0) as myCPX:

    dido = myCPX.modules[1] # change to your setup 

    for i in range(10):
        dido.set_channel(0)
        time.sleep(0.5)
        print(dido.read_channels()) 
        # should return a list of 8 items, the 5th being True. 
        # This is because the 4 inputs are returned first and after that the 4 outputs. 
        # You can also return only the outputs with read_channels(outputs_only=True). 
        # But this will be changed in the next release so you can use read_output_channels().

        dido.clear_channel(0)
        time.sleep(0.5)
        print(dido.read_channels()) # should return [False] * 8
    
```

**Comment by jhildenbrand-festo** [comment]
***In GitLab by @mw_festo on Aug 22, 2024, 08:00 UTC:***

I just saw that you are working with the non-released v0.5.4 that already implements the `read/write_output_channels` function. We just released these changes in v0.6.0, feel free to upgrade with `pip install festo-cpx-io --upgrade`
